src/randoop/JunitFileWriter.java

```diff
@@ -195,58 +183,86 @@ public File writeDriverFile(String driverClassName) {
     return junitTestSuites;
   }
   
-  public static File writeDriverFile(File dir, String packageName, String driverClassName,
-      List<String> junitTestSuiteNames) {
-    File file = new File(dir, driverClassName + ".java");
+  /*
+   * writes JUnit4 suite based on list of class names
+   */
+  public File writeSuiteFile (List<String> testClassNames) {
+    File dir = this.getDir();
+    String suiteClassName = masterTestClassName;
+    File file = new File(dir, suiteClassName+".java");
     PrintStream out = createTextOutputStream(file);
     try {
-      outputPackageName(out, packageName);
-      out.println("import junit.framework.*;");
-      out.println("import junit.textui.*;");
-      out.println("");
-      out.println("public class " + driverClassName + " extends TestCase {");
-      out.println("");
-      if (GenInputsAbstract.junit_reflection_allowed) {
-      out.println("  public static void main(String[] args) {");
-      } else {
-      out.println("  public static void main(String[] args) throws Throwable {");
+      outputPackageName(out,packageName);
+      
+      out.println();
+      out.println("import org.junit.runner.RunWith;");
+      out.println("import org.junit.runners.Suite;");
+      out.println();
+      out.println("@RunWith(Suite.class)");
+      out.println("@Suite.SuiteClasses({");
+     
+      Iterator<String> testIterator = testClassNames.iterator();
+      if (testIterator.hasNext()) {
+        out.println(testIterator.next() + ".class");
+        while (testIterator.hasNext()) {
+          out.print(", ");
+          out.println(testIterator.next() + ".class");
+        }
       }
-      if (GenInputsAbstract.init_routine != null)
-        out.println ("    " + GenInputsAbstract.init_routine + "();");
-
-      if (GenInputsAbstract.junit_reflection_allowed) {
-      out.println("    TestRunner runner = new TestRunner();");
-      out.println("    TestResult result = runner.doRun(suite(), false);");
-      out.println("    if (! result.wasSuccessful()) {");
-      out.println("      System.exit(1);");
-      out.println("    }");
-      } else {
-      out.println("    new " + driverClassName + "(\"" + driverClassName + "\").runTest();");
+      
+      out.println("})");
+      out.println("public class " + suiteClassName + "{ }");
+    } finally {
```

Comment:
I think a try-with-resources statement would be shorter and easier to read than the try-finally statement.
It's supported since Java 7, so it is fine to use.  (We are still avoiding use of Java 8 because some users haven't updated, but Java 7 is fair game.)